languages/haskell/src/DMN/Translate/PY.hs

```diff
@@ -0,0 +1,142 @@
+                               HP_Unique    -> "elif"
+                               HP_First     -> "elif"
+                               HP_Priority  -> "elif"
+                               HP_Collect _ -> "if"
+                               _            -> "if")
+  
+mkFunction tablename = [ "def", underscore tablename ]  -- helper function for function boilerplate 
+
+-- helper function for generating (arguments)
+mkArguments :: PYOpts -> [ColHeader] -> [String]
+mkArguments pyopts chs = ["(", intercalate ", " (mkArgument pyopts <$> input_headers chs), ")"]
+
+-- helper function for generating individual arguments that go into (args)
+mkArgument :: PYOpts -> ColHeader -> String
+mkArgument pyopts ch = var_name ch ++ maybe "" (const "") (vartype ch) 
+
+type2py :: DMNType -> String
+type2py DMN_String    = "str"
+type2py DMN_Number    = "float"
+type2py DMN_Boolean   = "bool"
+type2py (DMN_List x)  = type2py x ++ "[]"
+
+mkIf :: PYOpts -> HitPolicy -> String -> [ColHeader] -> DTrow -> String
+mkIf jsopts hp ifword chs dtrow =
+  let conditions = uncurry (fexp2js jsopts) <$> catMaybes ( zipWith nonBlankCols (input_headers chs) (row_inputs dtrow) )
+  in
+    "  " ++ ifword ++ " (" ++
+    (if not (null conditions)
+     then intercalate " and " conditions
+     else "\"default\"") -- TODO: tweak ifword to allow just an "else" here, rather than exploiting the truthiness of JS
+    ++ "): # " ++
+    maybe "cont'd" show (row_number dtrow) ++ "\n" ++
+    (let feelout = feel2pyOut hp chs dtrow
+         standard = maybe "" (\infra -> "    " ++ infra ++ "\n") (fst feelout) ++ "    return {" ++ intercalate ", " (snd feelout) ++ "};"
+     in
+     case hp of
+       HP_Unique    -> standard
+       HP_Any       -> standard
+       HP_Collect _ -> standard
+       HP_First     -> standard
+       HP_Priority  -> standard
+       HP_OutputOrder -> standard
+       HP_RuleOrder -> standard
+       HP_Aggregate -> standard
+    )
+    ++ "\n"
+    ++ annotationsAsComments chs dtrow
+
+-- if the row has multiple annotation columns, show the varname of the column header.
+-- if there is only one visible annotation column, hide the varname of the column header.
+annotationsAsComments :: [ColHeader] -> DTrow -> String
+annotationsAsComments chs dtrow =
+  let prefixedComments = catMaybes $ zipWith (\cheader commentcol -> ((varname cheader ++ ": ") ++) <$> commentcol) (comment_headers chs) (row_comments dtrow)
+      unprefixed = catMaybes $ row_comments dtrow
+  in
+  unlines $ ("    # "++) <$> (if length unprefixed > 1 then prefixedComments else unprefixed)
+
+fexp2js :: PYOpts -> ColHeader -> [FEELexp] -> String
+fexp2js jsopts ch fexps = wrapParen " or " (feel2pyIn ( showVarname jsopts ch) <$> fexps)
+
+showVarname :: PYOpts -> ColHeader -> String
+showVarname jsopts ch
+  | propstyle jsopts = "props[\"" ++ varname ch ++ "\"]"
+  | otherwise        = var_name ch
+               
+wrapParen :: String -> [String] -> String
```

Comment:
there are a number of functions which seem to be the same across JS.hs and PY.hs, like wrapParen, nonBlankCols, input_headers, comment_headers, etc. would it make more sense to put them into FEELhelpers so they don't get repeated?